platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/metadata/ComponentFileArtifactIdentifierSerializer.java

```diff
@@ -34,25 +32,13 @@ public class ComponentFileArtifactIdentifierSerializer implements Serializer<Com
     @Override
     public void write(Encoder encoder, ComponentFileArtifactIdentifier value) throws Exception {
         componentIdentifierSerializer.write(encoder, value.getComponentIdentifier());
-        Object rawFileName = value.getRawFileName();
-        if (rawFileName instanceof IvyArtifactName) {
-            encoder.writeBoolean(true);
-            IvyArtifactNameSerializer.INSTANCE.write(encoder, (IvyArtifactName) rawFileName);
-        } else {
-            encoder.writeBoolean(false);
-            encoder.writeString((String) rawFileName);
-        }
+        encoder.writeString(value.getFileName());
```

Comment:
❓ I think changing this causes the caches to become incompatible across versions and would require us to bump the cache version in `CacheLayout`. 

I think this doesn't show up in normal testing because you have to write with old/new versions of Gradle and switch between them for the same dependencies.

If I'm reading this correctly, I think you can retain compatibility by always writing a false boolean before writing the string.  This will force the old code to read the file name as-is.

Similarly, below, you'll need to still read the IvyArtifactName when the boolean is true.

Reply:
Nevermind, this isn't used by our caching.  It's only used by CC

Reply:
Discussed. This is only used for CC.